### PR TITLE
[fix](OrcWriter) fix be core when upgrading BE without upgrading FE

### PR DESCRIPTION
**Pull request by BePPPower**
Since we have changed the type mapping from Doris to Orc type, using the `Outfile` to export Date/Datetime types will cause BE core dump when only upgrading BE without upgrading FE.

### CODE REVIEW

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by BePPPower** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

File: gensrc/thrift/DataSinks.thrift [hunk 1]
```diff
@@ -133,6 +133,8 @@ struct TResultFileSinkOptions {
     18: optional bool with_bom;
 
     19: optional PlanNodes.TFileCompressType orc_compression_type;
+
+    20: optional i64 orc_writer_version;
```
Comment (morningman) [inline gensrc/thrift/DataSinks.thrift:137]: Please add some coment to explain this

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by morningman** [comment]
run buildall

**Comment by hello-stephen** [comment]
run buildall

**Review by CalvinKirs — APPROVED** [review]

**Review by hello-stephen — APPROVED** [review]

**Pull request merged**